File: my_py_pkg/my_py_pkg/number_counter.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int16

class NumberCounterNode(Node):
    def __init__(self):
        super().__init__("number_counter")
        self.subscriber_=self.create_subscription(Int16, "number",self.callback_number,10)
        self.get_logger().info("number Counter has started !")
        self.publisher_=self.create_publisher(Int16, "number_count",10)
        self.count=0

    def callback_number(self, msg: Int16):
        self.count+=msg.data
        new_msg = Int16()
        new_msg.data= self.count
        self.publisher_.publish(new_msg)

def main(args=None):
    rclpy.init(args=args)
    node= NumberCounterNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__=="__main__":
    main()

File: ros2_ws/src/my_cpp_pkg/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::placeholders;

class NumberCounterNode: public rclcpp::Node{
    public:
        NumberCounterNode(): Node("number_counter"), counter_(0)
        {
            number_subscriber_= this->create_subscription<example_interfaces::msg::Int64>("number", 10, std::bind(&NumberCounterNode::numberCallback, this, _1));
            counter_publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number_count", 10);

        }
    private:
        void numberCallback(const example_interfaces::msg::Int64::SharedPtr msg)
        {
            counter_ += msg->data;
            auto newMsg = example_interfaces::msg::Int64();
            newMsg.data = counter_;
            RCLCPP_INFO(this->get_logger(), "Got the number from number_publisher: %ld", msg->data);
            counter_publisher_->publish(newMsg);
        }
        int counter_;
        rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr counter_publisher_;
        rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr number_subscriber_;
};
int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberCounterNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}